Dockerfile can also build redis modules

lilatomic opened the pull request:
Redis includes several modules. The standard image does not include them.
This MR adds the capability for the Dockerfile to also build with modules. This allows for building images with or without the modules. 

relates to https://github.com/OT-CONTAINER-KIT/redis-operator/issues/1560

lilatomic committed:
reformat with heredoc

Signed-off-by: Daniel Goldman <danielgoldman4@gmail.com>

lilatomic committed:
add build deps and flags from docker library/redis

specifically https://github.com/redis/docker-library-redis/blob/a13b78815d980881e57f15b9cf13cd2f26f3fab6/alpine/Dockerfile

Signed-off-by: Daniel Goldman <danielgoldman4@gmail.com>

lilatomic committed:
default to not building with modules

Signed-off-by: Daniel Goldman <danielgoldman4@gmail.com>

lilatomic committed:
add to makefile

Signed-off-by: Daniel Goldman <danielgoldman4@gmail.com>

drivebyer commented on Dockerfile:
```diff
@@ -47,6 +60,7 @@ RUN apk upgrade --no-cache
 
 COPY --from=builder /usr/local/bin/redis-server /usr/local/bin/redis-server
 COPY --from=builder /usr/local/bin/redis-cli /usr/local/bin/redis-cli
+COPY --from=builder /tmp/redis-stable/modules/*/*.so /modules/
```
/tmp/redis-stable/modules/*/*.so should like /tmp/redis-${VERSION}/modules/*/*.so

raoganeshr commented:
@lilatomic are you planning to update as per review comments and submit?

lilatomic commented:
ah, I missed this! I've made the change and am testing it now.

lilatomic committed:
fix: use directory for version of redis being installed

Signed-off-by: Daniel Goldman <danielgoldman4@gmail.com>

lilatomic committed:
fix: allow building more recent redis versions

Redis > 8.2 wasn't building correctly.
Initially I was hitting an error about `if let`:

> `let` expressions in this position are unstable

in the `wildcard` transitive dependency of redisearch.

I upgraded the version of alpine used to get a more recent Rust toolchain which stabilised that syntax.

This caused another error because the llvm-dev package dropped the `/usr/bin/llvm-config` binary,
instead only including the binary with the version attached. (issue)[https://gitlab.alpinelinux.org/alpine/aports/-/work_items/17773].
The fix for that is in, but I wasn't getting it in the package archive.
So I added a check that sets the correct envvar.

Signed-off-by: Daniel Goldman <danielgoldman4@gmail.com>

lilatomic commented:
I hit some problems building the latest version of redis ("stable" is fine, 8.4 was not). I added the fixes in the last commit. You or I could drop/cherry-pick the commit if you'd rather that happen in a separate MR.

safaci2000 commented:
This would be handy to have. Any updates on merging this in or a fix to allow modules?
